packages/react/react/src/hooks/useLazyImage.ts

```diff
@@ -0,0 +1,67 @@
+import { useState } from 'react';
+import { useRefEffect } from './useRefEffect';
+
+interface Props {
+  src: string;
+  threshold?: number | number[];
+  root?: Document | Element | null;
+  rootMargin?: string;
+  onInView?: () => void;
+}
+
+/** @tossdocs-ignore */
+export function useLazyImage({ src, rootMargin, threshold, root, onInView }: Props) {
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  const ref = useRefEffect<HTMLImageElement>(element => {
+    if (typeof IntersectionObserver === 'undefined') {
+      element.src = src;
+      return;
+    }
+
+    if (element.getAttribute('src')) {
+      console.warn('If the "src" attribute is initially in an "img" tag, lazy load is not applied.');
+      return;
+    }
+
+    const insertImageSrc = ([entry]: IntersectionObserverEntry[], observer: IntersectionObserver) => {
+      if (entry) {
+        if (entry.isIntersecting) {
+          const imgElement = entry.target as HTMLImageElement;
+          imgElement.src = src;
+
+          imgElement.onload = () => {
+            setIsLoading(false);
+          };
+
+          /**
+           * Execute additional actions when the target element is exposed in the Viewport (or the element you specified as root)
+           * For example, logging processing
+           */
+          if (onInView) {
+            onInView();
+          }
+
+          /**
+           * Once the image has been loaded once, unobserve its target element to prevent repeated load.
+           */
+          observer.unobserve(entry.target);
+        }
+      }
+    };
+
+    const observer = new IntersectionObserver(insertImageSrc, {
+      root,
+      rootMargin,
+      threshold,
+    });
+
+    observer.observe(element);
+
+    return () => {
+      observer.unobserve(element);
+    };
+  }, []);
```

Comment:
I think deps should be added.
- src
- root
- onInView
- ...

I recommend you not to use `useRefEffect` because eslint error is ignored. Use `useEffect` instead:
```typescript
const ref = useRef<HTMLImageElement>(null);

useEffect(() => {
  const image = ref.current;
  if (image == null) {
    return;
  }

  image.src = src;
  // ...
}, [src, /* ... */]);
```

Reply:
I get it. I'll fix that.